lib/Controller/ApiController.php

```diff
@@ -73,12 +73,13 @@ public function __construct(IRequest $request,
 	 *
 	 * Set the download limit for a given link share
 	 */
-	public function setDownloadLimit(string $token, int $limit): Response {
+	public function setDownloadLimit(string $token, $limit): Response {
 		$this->validateToken($token);
+		$limit = trim($limit);
 
-		// Count needs to be at least 1
-		if ($limit < 1) {
-			throw new OCSBadRequestException('Limit needs to be greater or equal than 1');
+		// Count needs to be at least 0 or null
+		if (!is_numeric($limit) && $limit !== '') {
```

Comment:
```suggestion
		if ($limit >= 0) {
```

when keeping the int type on $limit

Reply:
You can use :
`if (($limit !== null) && ($limit < 0)) {`